### ISSUE TITLE
~ in config should be interpreted as HOME 

### ISSUE THREAD

**Issue by bestander** [yarnpkg/yarn#3393]
**Do you want to request a *feature* or report a *bug*?**
*feature*

**What is the expected behavior?**

```
yarn config set cache-folder ~/yarn-caches
```
Should create a folder in HOME/yarn-caches and use it for caches.
Currently ~ is not interpreted as HOME and yarn just creates folder named `~` in CWD of yarn execution.

This will be useful for multiple Yarn commands, so it would be nice to make this generic.

**Comment by bestander** [comment]
Duplicate of https://github.com/yarnpkg/yarn/issues/2008

**Comment by bestander** [comment]
Contributions are very much welcome

**Comment by samjacobclift** [comment]
This seems to be fixed in later versions of yarn
```
➜  test yarn config set cache-folder ~/yarn-caches
yarn config v0.22.0
success Set "cache-folder" to "/Users/samclift/yarn-caches".
✨  Done in 0.05s.
```

or am I missing something :)

**Comment by bestander** [comment]
It translates `~` to home folder on the way into .yarnrc.
In .yarnrc it will be saved as `/Users/samclift/yarn-caches` which can't be shared among multiple users

**Comment by bestander** [comment]
I'd like ~ inside .yarnrc to be interpreted as home folder

**Comment by samjacobclift** [comment]
Ahh I see

**Comment by mjesun** [comment]
Hi, I'll be working on this.

**Comment by bestander** [comment]
@mjesun, consider making it generic so that other config settings could use it.

**Comment by mjesun** [comment]
Hi @bestander,

There is not much we can do here. [Tilde expansion](https://www.gnu.org/software/bash/manual/html_node/Tilde-Expansion.html) happens before we can intercept it; so when we read it there is no access to the original value. However, when you quote it (i.e., `yarn config set cache-folder '~/yarn-caches'`) it works.

**Comment by bestander** [comment]
Yeah, we can cover this case.
I am more interested in ~ expansion at read time

**Comment by mjesun** [comment]
Ok, sounds cool :)
